src/main/java/com/blackducksoftware/integration/hub/alert/channel/DistributionChannel.java

```diff
@@ -22,19 +22,82 @@
+import com.blackducksoftware.integration.hub.notification.processor.ItemTypeEnum;
+import com.blackducksoftware.integration.hub.notification.processor.NotificationCategoryEnum;
 import com.google.gson.Gson;
 
 public abstract class DistributionChannel<E extends AbstractChannelEvent, C extends DatabaseEntity> extends MessageReceiver<E> {
+    private final static Logger logger = LoggerFactory.getLogger(DistributionChannel.class);
+    protected final JpaRepository<C, Long> repository;
 
-    public DistributionChannel(final Gson gson, final Class<E> clazz) {
+    public DistributionChannel(final Gson gson, final JpaRepository<C, Long> repository, final Class<E> clazz) {
         super(gson, clazz);
+        this.repository = repository;
     }
 
     public abstract void sendMessage(final E event, final C config);
 
     public abstract String testMessage(final C config);
 
+    protected String createHtmlMessage(final ProjectData projectData) {
+        final StringBuilder htmlBuilder = new StringBuilder();
+        htmlBuilder.append("<strong>" + projectData.getProjectName() + " > " + projectData.getProjectVersion() + "</strong>");
+
+        final Map<NotificationCategoryEnum, CategoryData> categoryMap = projectData.getCategoryMap();
+        if (categoryMap != null) {
+            for (final NotificationCategoryEnum category : NotificationCategoryEnum.values()) {
+                final CategoryData data = categoryMap.get(category);
+                if (data != null) {
+                    htmlBuilder.append("<br />- - - - - - - - - - - - - - - - - - - -");
+                    htmlBuilder.append("<br />Type: " + data.getCategoryKey());
+                    htmlBuilder.append("<br />Number of Changes: " + data.getItemCount());
+                    for (final ItemData item : data.getItemList()) {
+                        final Map<String, Object> dataSet = item.getDataSet();
+                        htmlBuilder.append("<p>  Rule: " + dataSet.get(ItemTypeEnum.RULE.toString()));
+                        htmlBuilder.append(" | Component: " + dataSet.get(ItemTypeEnum.COMPONENT.toString()));
+                        htmlBuilder.append(" [" + dataSet.get(ItemTypeEnum.VERSION.toString()) + "]</p>");
+                    }
+                }
+            }
+        } else {
+            htmlBuilder.append("<br /><i>A notification was received, but it was empty.</i>");
+        }
+        return htmlBuilder.toString();
+    }
+
+    protected String createPlaintextMessage(final ProjectData projectData) {
+        final String htmlMessage = createHtmlMessage(projectData);
+        return StringEscapeUtils.escapeHtml4(htmlMessage);
+    }
+
+    public void handleEvent(final E event) {
+        final List<C> configurations = repository.findAll();
+        for (final C configEntity : configurations) {
+            sendMessage(event, configEntity);
+        }
+    }
+
+    @Override
+    public void receiveMessage(final String message) {
```

Comment:
This method may make sense here given that handleEvent will be abstract and the individual channel can determine what to do with the event.